common/state_of_charge.hpp

```diff
@@ -0,0 +1,57 @@
+#include "devices/sensors/battery/max17043.hpp"
+#include "peripherals/hardware_counter.hpp"
+#include "utility/log.hpp"
+namespace sjsu::common
+{
+
+  class StateOfCharge
+  {
+    public:
+    sjsu::Max17043 &module_;
+    StateOfCharge(sjsu::Max17043 &module) : module_(module) {}
+
+    void Initialize()
+    {
+      module_.ModuleInitialize();
+    }
+
+    double BatteryPercentage()
+    {
+      units::voltage::volt_t battery_voltage = module_.GetVoltage();
+      double voltage_percent = double(battery_voltage);
+      voltage_percent = (voltage_percent - 3) / (4.2 - 3);
```

Comment:
Not sure where you got these numbers from? Do they mean something? Might want to create variables for them